contact/forms.py

```diff
@@ -30,6 +31,11 @@ def __init__(self, *args, **kwargs):
         for field in self.required_fields:
             self.fields[field].required = True
 
+        # Update field placeholder
+        for field, placeholder in self.field_placeholders.items():
+            field = self.fields.get(field)
+            field.widget.attrs["placeholder"] = placeholder
```

Comment:
Will this error if it doesn't find a field and `field` ends up being `None`?

Reply:
So it only errors if you enter a field which isn't present in the form. If you set no fields then it'll just use the empty dict and not throw any errors (at least from my testing)

Reply:
@Will-Hoey All good. Just making sure that was the plan.